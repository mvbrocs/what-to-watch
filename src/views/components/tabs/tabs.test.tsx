import React from 'react';
import { mount } from 'enzyme';

import { Tabs } from './';
import { TabPane } from './';

describe('Tabs', () => {
  const tabs = mount(
    <Tabs>
      <TabPane tab="Tab 1" index="1">
        Tab content 1
      </TabPane>
      <TabPane tab="Tab 2" index="2">
        Tab content 2
      </TabPane>
    </Tabs>,
  );

  it('renders properly', () => {
    expect(tabs).toMatchSnapshot();
  });

  it('should correctly toggle tabs', () => {
    tabs
      .find('.movie-nav__link')
      .at(1)
      .simulate('click', { preventDefault: () => {} });

    expect(
      tabs
        .find('.movie-nav__item')
        .at(0)
        .prop('className'),
    ).toEqual('movie-nav__item');
    expect(
      tabs
        .find('.movie-nav__item')
        .at(1)
        .prop('className'),
    ).toEqual('movie-nav__item movie-nav__item--active');
    expect(
      tabs
        .find('[index="1"]')
        .parent()
        .prop('hidden'),
    ).toBeTruthy();
    expect(
      tabs
        .find('[index="2"]')
        .parent()
        .prop('hidden'),
    ).toBeFalsy();
  });
});
