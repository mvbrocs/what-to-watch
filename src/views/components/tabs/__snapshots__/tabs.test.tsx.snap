// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders properly 1`] = `
<Fragment>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item movie-nav__item--active"
        key="1"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Tab 1
        </a>
      </li>
      <li
        className="movie-nav__item"
        key="2"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Tab 2
        </a>
      </li>
    </ul>
  </nav>
  <div
    hidden={false}
    key="1"
  >
    <TabPane
      index="1"
      tab="Tab 1"
    >
      Tab content 1
    </TabPane>
  </div>
  <div
    hidden={true}
    key="2"
  >
    <TabPane
      index="2"
      tab="Tab 2"
    >
      Tab content 2
    </TabPane>
  </div>
</Fragment>
`;
